Enable ORY Hydra Maester in Kyma (#5480)
kyma-project/kyma: docs/security/03-06-oauth2-server.md

Out of the box, the Kyma implementation of the ORY stack supports the [OAuth 2.0 Client Credentials Grant](https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/).

>**NOTE:** The implementation of the ORY Oauth2 server in Kyma is still in the early stages and is subject to changes and development. Read [this](https://kyma-project.io/blog/2019/7/31/kyma-collaboration-with-ory/) blog post to get a better understanding of the target integration of the ORY stack in Kyma.

## Register an OAuth2 client

To interact with the Kyma OAuth2 server, you must register an OAuth2 client. Run:

```
cat <<EOF | kubectl apply -f -
apiVersion: hydra.ory.sh/v1alpha1
kind: OAuth2Client
metadata:
  name: my-oauth2-client
  namespace: my-namespace
spec:
  grantTypes:
    - "client_credentials"
  scope: "read write"
EOF
```

As a result, Hydra Maester sends a registration request to the OAuth2 server and saves incoming credentials to a Kubernetes Secret.
>**CAUTION:** The instance of the`oauth2clients.hydra.ory.sh` custom resource and its corresponding Secret share a name and Namespace.

> To retrieve your client credentials granted by the OAuth2 sever, run this command:

Reviewer: ```suggestion
Run this command to get the credentials of the registered OAuth2:
```

```

> kubectl get secret -n my-namespace my-oauth2-client -o yaml

Reviewer: ```suggestion
kubectl get secret -n {CLIENT_NAMESPACE} {NAME_OF_CLIENT} -o yaml
```

```

> Visit [ORY Helm Charts](https://github.com/ory/k8s/blob/master/docs/helm/hydra-maester.md) Github page to learn more about the `oauth2clients.hydra.ory.sh` custom resource.

Reviewer: ```suggestion
See [this](https://github.com/ory/k8s/blob/master/docs/helm/hydra-maester.md) ORY document to learn more about the `oauth2clients.hydra.ory.sh` custom resource.
```

> See Ory Hydra Maester [github page](https://github.com/ory/hydra-maester) to learn more about the `oauth2clients.hydra.ory.sh` custom resource.

Reviewer: ```suggestion
See the ORY Hydra Maester [github page](https://github.com/ory/hydra-maester) to learn more about the `oauth2clients.hydra.ory.sh` custom resource.
```
